litho-core/src/main/java/com/facebook/litho/LithoNode.java

```diff
@@ -185,6 +186,10 @@ public class LithoNode implements Node<LithoRenderContext>, Cloneable {
 
   protected @Nullable YogaMeasureFunction mYogaMeasureFunction;
 
+  protected @Nullable YogaGutter mGap;
+  protected @Nullable Integer mGapLength;
```

Comment:
We are in the process of generalising the implementation of LithoNode (i.e. decouple Litho and Yoga) so we don't recommend adding new Yoga props to LithoNode. Instead they should be implemented using the `LayoutProps` APIs.

New `gap`, and `gapLength` APIs can be added to the [LayoutProps](https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/LayoutProps.kt#L25C24-L25C24) interface. Then implement the new APIs in [CommonProps](https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/CommonProps.java) for the component to write to, and [YogaPropsWriter](https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/YogaLayoutProps.kt#L30) for the YogaNode to be written to.

The API can then be exposed to Row and Column component builders by adding new setters in (ContainerBuilder)[https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/Component.java#L2161]. In the implementation it would call the new API on the common props.

Reply:
Thanks so much for the feedback, I will give this a go and reach out if I need any support